hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -2669,9 +2671,7 @@ S3AFileStatus innerGetFileStatus(final Path f,
       // modification - compare the modTime to check if metadata is up to date
       // Skip going to s3 if the file checked is a directory. Because if the
       // dest is also a directory, there's no difference.
-      // TODO After HADOOP-16085 the modification detection can be done with
```

Comment:
If we really do modtime that's bad. That's so bad because we never store the modtime right when a new file is created, so we query it two times! Something to review and work on! (I'm not sure about this, this is just a fyi, should be checked.)

Reply:
see https://issues.apache.org/jira/browse/HADOOP-16644 ; if there's any clock skew then the first open/head triggers a s3guard update, which makes Sidd sad. For a simple S3 PUT we know the modtime; for a multipart we can just do one more HEAD request at create time

Reply:
ok, this should be handled as a separate case then.

Reply:
+1